Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1226,11 +1226,22 @@ extension BrowserViewController: URLBarDelegate {
         })
         longPressAlertController.addAction(cancelAction)
 
-        if let popoverPresentationController = longPressAlertController.popoverPresentationController {
-            popoverPresentationController.sourceView = urlBar
-            popoverPresentationController.sourceRect = urlBar.frame
-            popoverPresentationController.permittedArrowDirections = .Any
+        let setupPopover = { [unowned self] in
+            if let popoverPresentationController = longPressAlertController.popoverPresentationController {
+                popoverPresentationController.sourceView = urlBar
+                popoverPresentationController.sourceRect = urlBar.frame
+                popoverPresentationController.permittedArrowDirections = .Any
+                popoverPresentationController.delegate = self
+            }
+        }
+
+        setupPopover()
+
+        if longPressAlertController.popoverPresentationController != nil {
+            displayedPopoverController = longPressAlertController
```

Comment:
I think we want `displayedPopoverController` to be `weak`, right? Since we never set it to `nil`, we'll hold onto it even after it has been dismissed.